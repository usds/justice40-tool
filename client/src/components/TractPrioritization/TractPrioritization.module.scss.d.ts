declare namespace TractPrioritizationNamespace {
    export interface ITractPrioritizationScss {
      invert: string;
    }
  }

declare const TractPrioritizationScssModule: TractPrioritizationNamespace.ITractPrioritizationScss & {
    /** WARNING: Only available when "css-loader" is used without "style-loader" or "mini-css-extract-plugin" */
    locals: TractPrioritizationNamespace.ITractPrioritizationScss;
  };

  export = TractPrioritizationScssModule;
