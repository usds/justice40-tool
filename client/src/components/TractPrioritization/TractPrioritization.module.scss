@use '../../styles/design-system.scss' as *;
@import "../utils.scss";

h3.invert{
    @include invert();
};
