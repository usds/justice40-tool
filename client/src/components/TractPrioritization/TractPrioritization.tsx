import React from 'react';

import * as styles from './TractPrioritization.module.scss';

import * as EXPLORE_COPY from '../../data/copy/explore';

interface ITractPrioritization {
  scoreNCommunities: boolean,
  tribalCountAK: number | null,
  tribalCountUS: number | null,
  percentTractTribal: number | null,
}

/**
 * This component will return the appropriate designation for the tract's prioritization
 *
 * @param {boolean} scoreNCommunities
 * @param {number | null} tribalCountAK
 * @param {number | null} tribalCountUS
 * @param {number | null} percentTractTribal
 * @return {JSX}
 */
const TractPrioritization = (
    {scoreNCommunities,
      tribalCountAK,
      tribalCountUS,
      percentTractTribal}:ITractPrioritization) => {
  if (scoreNCommunities === true) {
    return <h3 className={styles.invert}>{EXPLORE_COPY.COMMUNITY.OF_FOCUS}</h3>;
  } else {
    if (percentTractTribal === null && tribalCountAK === null && tribalCountUS === null) {
      return <h3>{EXPLORE_COPY.COMMUNITY.NOT_OF_FOCUS}</h3>;
    } else {
      return <h3 className={styles.invert}>{EXPLORE_COPY.COMMUNITY.PARTIAL}</h3>;
    }
  }
};

export default TractPrioritization;
