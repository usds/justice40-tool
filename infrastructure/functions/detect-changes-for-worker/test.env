REGION=us-east-1
STAGE=sit

# ESC Cluster name that will from the containers
ECS_CLUSTER=j40-sit-justice40-data-harvester-ECSCluster-ktXGGU9zjwkb

# VPC Private Subnet that has a NAT GAteway
VPC_SUBNET_ID=subnet-07e68cb57322f7b1f

# Names of the container and task names.
GDAL_TASK_DEFINITION=sit-justice40-data-harvester-gdal
GDAL_CONTAINER_DEFINITION=sit-justice40-data-harvester-osgeo-gdal
TIPPECANOE_TASK_DEFINITION=sit-justice40-data-harvester-tippecanoe
TIPPECANOE_CONTAINER_DEFINITION=sit-justice40-data-harvester-mgiddens-tippecanoe

# AWS Credentials
AWS_ACCESS_KEY_ID=
AWS_SECRET_ACCESS_KEY=
AWS_DEFAULT_REGION=us-east-1