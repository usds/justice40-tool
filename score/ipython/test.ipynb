{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "1a4c0c68",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "70b3a793",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv('data/fips_states.csv')  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "c514aad8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>fips</th>\n",
       "      <th>state_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>Alabama</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>Alaska</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4</td>\n",
       "      <td>Arizona</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5</td>\n",
       "      <td>Arkansas</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>6</td>\n",
       "      <td>California</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   fips  state_name\n",
       "0     1    Alabama \n",
       "1     2     Alaska \n",
       "2     4    Arizona \n",
       "3     5   Arkansas \n",
       "4     6  California"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b9ee44d9",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
