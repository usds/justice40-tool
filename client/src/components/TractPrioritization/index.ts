import TractPrioritization from './TractPrioritization';
export default TractPrioritization;
