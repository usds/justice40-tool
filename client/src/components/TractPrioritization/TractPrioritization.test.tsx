import React from 'react';
import {render, screen} from '@testing-library/react';
import {LocalizedComponent} from '../../test/testHelpers';
import TractPrioritization from './TractPrioritization';

describe('rendering of TractPrioritization Component', () => {
  const testCases = [
    {scoreNComm: true, tribalCountAK: null, percentTribal: null, isDAC: 'YES'},
    {scoreNComm: true, tribalCountAK: null, percentTribal: 0, isDAC: 'YES'},
    {scoreNComm: true, tribalCountAK: null, percentTribal: .45, isDAC: 'YES'},
    {scoreNComm: true, tribalCountAK: 1, percentTribal: null, isDAC: 'YES'},
    {scoreNComm: true, tribalCountAK: 2, percentTribal: 0, isDAC: 'YES'},
    {scoreNComm: true, tribalCountAK: 3, percentTribal: .45, isDAC: 'YES'},

    {scoreNComm: false, tribalCountAK: null, percentTribal: null, isDAC: 'NO'},
    {scoreNComm: false, tribalCountAK: null, percentTribal: 0, isDAC: 'PARTIALLY'},
    {scoreNComm: false, tribalCountAK: null, percentTribal: .13, isDAC: 'PARTIALLY'},
    {scoreNComm: false, tribalCountAK: 1, percentTribal: null, isDAC: 'PARTIALLY'},
    {scoreNComm: false, tribalCountAK: 2, percentTribal: 0, isDAC: 'PARTIALLY'},
    {scoreNComm: false, tribalCountAK: 3, percentTribal: .13, isDAC: 'PARTIALLY'},

  ];

  testCases.forEach((testCase) => {
    // eslint-disable-next-line max-len
    it(`checks if component renders ${testCase.isDAC} when score N communities = ${testCase.scoreNComm}, tribal points in AK = ${testCase.tribalCountAK} tribal % = ${testCase.percentTribal}`, () => {
      const {asFragment} = render(
          <LocalizedComponent>
            <TractPrioritization
              scoreNCommunities={testCase.scoreNComm}
              tribalCountAK={testCase.tribalCountAK}
              tribalCountUS={null}
              percentTractTribal={testCase.percentTribal}
            />
          </LocalizedComponent>,
      );
      expect(asFragment()).toMatchSnapshot();
      screen.getByText(testCase.isDAC);
    });
  });
});
