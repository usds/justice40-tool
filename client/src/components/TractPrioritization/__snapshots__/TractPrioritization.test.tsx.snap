// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering of TractPrioritization Component checks if component renders NO when score N communities = false, tribal points in AK = null tribal % = null 1`] = `
<DocumentFragment>
  <h3>
    NO
  </h3>
</DocumentFragment>
`;

exports[`rendering of TractPrioritization Component checks if component renders PARTIALLY when score N communities = false, tribal points in AK = 1 tribal % = null 1`] = `
<DocumentFragment>
  <h3>
    PARTIALLY
  </h3>
</DocumentFragment>
`;

exports[`rendering of TractPrioritization Component checks if component renders PARTIALLY when score N communities = false, tribal points in AK = 2 tribal % = 0 1`] = `
<DocumentFragment>
  <h3>
    PARTIALLY
  </h3>
</DocumentFragment>
`;

exports[`rendering of TractPrioritization Component checks if component renders PARTIALLY when score N communities = false, tribal points in AK = 3 tribal % = 0.13 1`] = `
<DocumentFragment>
  <h3>
    PARTIALLY
  </h3>
</DocumentFragment>
`;

exports[`rendering of TractPrioritization Component checks if component renders PARTIALLY when score N communities = false, tribal points in AK = null tribal % = 0 1`] = `
<DocumentFragment>
  <h3>
    PARTIALLY
  </h3>
</DocumentFragment>
`;

exports[`rendering of TractPrioritization Component checks if component renders PARTIALLY when score N communities = false, tribal points in AK = null tribal % = 0.13 1`] = `
<DocumentFragment>
  <h3>
    PARTIALLY
  </h3>
</DocumentFragment>
`;

exports[`rendering of TractPrioritization Component checks if component renders YES when score N communities = true, tribal points in AK = 1 tribal % = null 1`] = `
<DocumentFragment>
  <h3>
    YES
  </h3>
</DocumentFragment>
`;

exports[`rendering of TractPrioritization Component checks if component renders YES when score N communities = true, tribal points in AK = 2 tribal % = 0 1`] = `
<DocumentFragment>
  <h3>
    YES
  </h3>
</DocumentFragment>
`;

exports[`rendering of TractPrioritization Component checks if component renders YES when score N communities = true, tribal points in AK = 3 tribal % = 0.45 1`] = `
<DocumentFragment>
  <h3>
    YES
  </h3>
</DocumentFragment>
`;

exports[`rendering of TractPrioritization Component checks if component renders YES when score N communities = true, tribal points in AK = null tribal % = 0 1`] = `
<DocumentFragment>
  <h3>
    YES
  </h3>
</DocumentFragment>
`;

exports[`rendering of TractPrioritization Component checks if component renders YES when score N communities = true, tribal points in AK = null tribal % = 0.45 1`] = `
<DocumentFragment>
  <h3>
    YES
  </h3>
</DocumentFragment>
`;

exports[`rendering of TractPrioritization Component checks if component renders YES when score N communities = true, tribal points in AK = null tribal % = null 1`] = `
<DocumentFragment>
  <h3>
    YES
  </h3>
</DocumentFragment>
`;
